source/link.cpp

```diff
@@ -0,0 +1,80 @@
+#include "link.hpp"
+
+#include "event.hpp"
+#include "scheduler.hpp"
+
+namespace sim {
+
+Link::Link(std::weak_ptr<IRoutingDevice> a_from,
+           std::weak_ptr<IRoutingDevice> a_to, std::uint32_t a_speed_mbps,
+           std::uint32_t a_delay)
+    : m_from(a_from),
+      m_to(a_to),
+      m_speed_mbps(a_speed_mbps),
+      m_transmission_delay(a_delay),
+      m_src_egress_delay(0) {
+    if (a_from.expired() || a_to.expired()) {
+        // TODO: warning log
+    } else if (a_speed_mbps == 0) {
+        // TODO: warning log
+    }
+}
+
+std::uint32_t Link::get_transmission_time(const Packet& packet) {
```

Comment:
Should we name function arguments like this?
```suggestion
std::uint32_t Link::get_transmission_time(const Packet& a_packet) {
```

Reply:
I dont think we should
This helps to avoid conflicts in constructors but I don't think we need it in methods